clang/test/CodeGen/X86/sse-builtins.c

```diff
@@ -6,13 +6,23 @@
 
 #include <immintrin.h>
 
+#if defined(__cplusplus) && (__cplusplus >= 201103L)
+constexpr bool match_m128(__m128 v, float x, float y, float z, float w) {
+  return v[0] == x && v[1] == y && v[2] == z && v[3] == w;
```

Comment:
Can we templatize it to make it more general?

Reply:
I'd prefer to keep the matching code simple if we can - since these tests are testing the clang C++ frontend we probably shouldn't make it anymore complicated than necessary.